src/view/document.js

```diff
@@ -256,6 +256,17 @@ export default class Document {
 	}
 
 	/**
+	 * Focuses document. It will focus {@link engine.view.EditableElement EditableElement} that is currently having
+	 * selection inside.
+	 *
+	 * @fires engine.view.Document#focus
+	 */
+	focus() {
+		this.fire( 'focus' );
```

Comment:
Also I am not fire if `focus` event should be fired if there already was focus. You could try to check if `isFocused` is false.